src/pstress.cpp

```diff
@@ -49,6 +52,20 @@ void create_worker(struct workerParams *Params) {
 }
 
 int main(int argc, char *argv[]) {
+
+  /* Extract the fullpath of the executable binary */
+  char buffer[100];
+  realpath(argv[0], buffer);
+  std::string a = buffer;
+  fullpath = std::string(a, 0, fullpath.rfind("/"));
+
+  /* Extract the binary name from the fullpath */
+  auto binary_name = fullpath.substr(fullpath.find_last_of("/\\") + 1);
+  std::string::size_type i = fullpath.find(binary_name);
+
+  /* Remove the binary name from the fullpath */
+  fullpath.erase(i, binary_name.length());
```

Comment:
May be you could just use `dirname()` function to get the directory name. See https://linux.die.net/man/3/dirname

Reply:
done